app/models/course.rb

```diff
@@ -0,0 +1,21 @@
+class Course < ApplicationRecord
+  belongs_to :teacher, class_name: "User"
+  belongs_to :category
+  has_many :lessons, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  has_one :quiz
+
+  has_one_attached :display_picture
+  has_one_attached :syllabus
+  has_one_attached :completion_certificate
+  has_one_attached :achievement_certificate
+
+  enum level: %i[beginner intermediate advanced].freeze
+
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :price, presence: true
```

Comment:
Also can Price be negative?

Reply:
No apu, price can't be negative. Added validation for price.

Reply:
Added validation apu.